packages/auto-label/src/auto-label.ts

```diff
@@ -247,94 +221,78 @@ handler.addLabeltoRepoAndIssue = async function addLabeltoRepoAndIssue(
     issueNumber
   );
 
+  let autoDetectedLabel: string | undefined;
+
   if (!objectInJsonArray?.github_label) {
     console.log(
       `There was no configured match for the repo ${repo}, trying to auto-detect the right label`
     );
-    const autoDetectedLabel = handler.autoDetectLabel(jsonArray, issueTitle);
-    if (autoDetectedLabel) {
-      console.log(
-        `Auto-detected label ${autoDetectedLabel} for ${owner}/${repo}#${issueNumber}`
-      );
-      autoDetectedLabelAdded = await handler.maybeAddAPILabel(
+    autoDetectedLabel = handler.autoDetectLabel(jsonArray, issueTitle);
+  }
+  const colorNumber =
+    jsonArray?.findIndex((object: JSONData) => objectInJsonArray === object) %
+      colorsData.length || 0;
+
+  const githubLabel = objectInJsonArray?.github_label || autoDetectedLabel;
+
+  let alreadyExists;
+  if (githubLabel) {
+    alreadyExists = await handler.checkExistingLabels(
```

Comment:
If I'm reading this right, it looks like you make an API call to check if a given issue has a label, and if it doesn't, you add it.  Instead - why not just always try to add the label?  As long as it has the same text/title, the GitHub API (I think) should figure it out, and it would reduce the complexity and number of API calls.

Reply:
good call!